core-bundle/src/EventListener/DataContainer/TemplateOptionsListener.php

```diff
@@ -45,10 +45,11 @@ public function __invoke(DataContainer $dc): array
             return array_merge(['' => '-'], $this->controller->getTemplateGroup($this->templatePrefix));
         }
 
-        $defaultTemplate = $this->customTemplates[$dc->activeRecord->type] ?? $this->getLegacyDefaultTemplate($dc);
+        $currentRecord = $dc->getCurrentRecord();
+        $defaultTemplate = $this->customTemplates[$currentRecord['type']] ?? $this->getLegacyDefaultTemplate($dc);
```

Comment:
do we need to check if `$currentRecord['type']` exists or are we sure this key is always defined (applies a few lines below as well)?

Reply:
We didn't check before either so I would keep it as is.

Reply:
yeah, previously it was an object, which does not trigger the invalid array key error, does it?

Reply:
https://3v4l.org/uZr0s#v8.1.7